### DISCUSSION TITLE
How to use nvim-treesitter-refactor with rocks-treesitter?

### DISCUSSION THREAD

**Discussion by kflak** [lumen-oss/rocks.nvim#325] [Q&A]
First time poster here! Thanks for an awesome package manager :-)

Question: I have a couple of functions from [nvim-treesitter-refactor](https://github.com/nvim-treesitter/nvim-treesitter-refactor) that I really like from my previous treesitter setup, and would like to bring them over to the rocks side. But how...?

**Comment by salkin-mada** [comment]
refactor is not in the laurock repo as of now. So using rocks-git? `Rocks install nvim-treesitter/nvim-treesitter-refactor`

**Reply by kflak** [reply]
Don't think so... the only way to access its settings is within the `nvim-treesitter` configuration, which would conflict with `rocks-treesitter`

**Reply by salkin-mada** [reply]
So how to add treesitter _module/extension_ to `rocks-treesitter` in general?
I have no idea. We have to ask @mrcjkb
<3

**Reply by mrcjkb** [reply]
Hey :wave:

It should technically be possible to use `nvim-treesitter` with `rocks-treesitter`. You just have to disable auto install and auto highlight in `nvim-treesitter`.
One thing that could be an issue is the fact that `nvim-treesitter`'s `master` branch comes with queries that may be outdated.
:thinking: A potential workaround for this could be to publish a `nvim-treesitter-legacy` luarock that excludes the queries, and provides just the legacy lua API.

There are plans to refactor `nvim-treesitter-refactor` into a standalone plugin: https://github.com/nvim-treesitter/nvim-treesitter-refactor/issues/53.

**Reply by mrcjkb** [reply]
You can try [`nvim-treesitter-legacy-api`](https://luarocks.org/modules/neorocks/nvim-treesitter-legacy-api)

**Reply by kflak** [reply]
Wow, that was fast!

Will check it out when I have a bit more time. Thanks :-)

**Reply by kflak** [reply]
OK, had a bit of a moment to test this, but it doesn't seem to have any effect. Using this config, which is copied pretty much as-is from my previous lazy setup (minus the lazy-specific bits, obviously)

```lua
require'nvim-treesitter.configs'.setup {
    ensure_installed =  {
        'bash',
        'c',
        'cmake',
        'comment',
        'commonlisp',
        'cpp',
        'css',
        'html',
        'javascript',
        'json',
        'lua',
        'make',
        'org',
        'python',
        'regex',
        'rust',
        'supercollider',
        'toml',
        'vim',
        'vimdoc',
        'yaml'
    }, -- could be 'all'   
    matchup = {
        enable = true,
    },
    sync_install = false,
    auto_install = true,
    -- ignore_install = { "commonlisp" },
    highlight = {
        enable = true,              -- false will disable the whole extension
        additional_vim_regex_highlighting = {'org'},
    },
    indent = {
        enable = {},
        -- disable = { 'supercollider' },
    },
    refactor = {
        smart_rename = {
            enable = true,
            keymaps = {
                smart_rename = "grr",
            },
        },
        navigation = {
            enable = true,
            -- Assign keymaps to false to disable them, e.g. `goto_definition = false`.
            keymaps = {
                goto_definition = "gnd",
                list_definitions = "gnD",
                list_definitions_toc = "gO",
                goto_next_usage = "<a-*>",
                goto_previous_usage = "<a-#>",
            },
        },
        highlight_current_scope = { enable = true },
    },
}
```

**Reply by mrcjkb** [reply]
I've just uploaded a new rockspec with a fix. It should work after reinstalling.

**Reply by kflak** [reply]
Cool, it works! Thanks a lot. Looking forward to when the refactor becomes a separate plugin, though... The current setup does feel a bit hacky.
